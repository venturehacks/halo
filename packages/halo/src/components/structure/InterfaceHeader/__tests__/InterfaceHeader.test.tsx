import { render } from 'enzyme';
import React from 'react';

import { InterfaceHeader } from '../index';

describe('InterfaceHeader', () => {
  test('smoke', () => {
    const component = render(
      <InterfaceHeader page>General Settings</InterfaceHeader>,
    );
    expect(component.text()).toContain('General Settings');
  });

  test('className', () => {
    const component = render(
      <InterfaceHeader className="mb-10" page>
        General Settings
      </InterfaceHeader>,
    );
    expect(component.hasClass('mb-10')).toBeTruthy();
  });

  describe('lacking hierarhcy', () => {
    test('throws exception ', () => {
      expect(() =>
        render(
          <InterfaceHeader>This should throw an exception</InterfaceHeader>,
        ),
      ).toThrow();
    });
  });

  describe('snapshot', () => {
    test('page', () => {
      const component = render(
        <InterfaceHeader page>General Settings</InterfaceHeader>,
      );

      expect(component).toMatchSnapshot();
    });

    test('panel', () => {
      const component = render(
        <InterfaceHeader panel>Unbiased Sourcing</InterfaceHeader>,
      );

      expect(component).toMatchSnapshot();
    });

    test('section', () => {
      const component = render(
        <InterfaceHeader section>Diversity</InterfaceHeader>,
      );

      expect(component).toMatchSnapshot();
    });

    test('micro', () => {
      const component = render(
        <InterfaceHeader micro>Ideal next opportunity</InterfaceHeader>,
      );

      expect(component).toMatchSnapshot();
    });
  });
});
