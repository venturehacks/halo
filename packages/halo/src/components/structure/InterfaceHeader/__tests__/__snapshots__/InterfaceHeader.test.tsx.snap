// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterfaceHeader snapshot micro 1`] = `
<header
  class="text-xs text-dark-aa font-normal"
>
  Ideal next opportunity
</header>
`;

exports[`InterfaceHeader snapshot page 1`] = `
<header
  class="text-xl text-slate-800 font-normal"
>
  General Settings
</header>
`;

exports[`InterfaceHeader snapshot panel 1`] = `
<header
  class="text-lg text-dark-aaa font-normal"
>
  Unbiased Sourcing
</header>
`;

exports[`InterfaceHeader snapshot section 1`] = `
<header
  class="text-md text-dark-aaa font-normal"
>
  Diversity
</header>
`;
